assets/js/plugins.coffee

```diff
@@ -0,0 +1,99 @@
+if module?
+  global._ = require('lodash')
+  global.Logger = require('./logger.coffee')
+  global.DependencyGraph = require('dependencies-online')
+
+(() ->
+  class PluginDatabaseAccess
+    constructor: (@data, @pluginName) ->
+
+    getVersion: () ->
+      @data.store.getPluginVersion @pluginName
+    setVersion: (version) ->
+      @data.store.setPluginVersion @pluginName, version
+    setRowData: (row, args...) ->
+      if args.length == 1
+        [key, [data]] = ['data', args]
+      else if args.length == 2
+        [key, data] = args
+      @data.store.setPluginRowData @pluginName, row.id, key, data
+    getRowData: (row, key='data') ->
+      @data.store.getPluginRowData @pluginName, row.id, key
+    setGlobalData: (key, args...) ->
```

Comment:
I originally wrote it without key; I think it's better this way. (The idea is basically that the plugin gets its own tiny global key-value store, as opposed to just one slot, so we can make writes/reads granular)